src/lightning_fabric/fabric.py

```diff
@@ -608,9 +609,17 @@ def log_dict(self, metrics: Dict[str, Any], step: Optional[int] = None) -> None:
 
         Args:
             metrics: A dictionary where the key is the name of the metric and the value the scalar to be logged.
+                Any :class:`torch.Tensor`s in the dictionary get detached from the graph automatically.
             step: Optional step number. Most Logger implementations auto-increment this value by one with every
                 log call. You can specify your own value here.
         """
+
+        def to_item(value: Tensor) -> int | float | bool:
+            if value.numel() != 1:
+                raise ValueError("Logging tensors with more than one element is not supported.")
+            return value.item()
```

Comment:
Should we pull `LightningModule.__to_tensor` to an utility function to use it here? It also detaches and squeezes

Reply:
We already have this in yet another utility function called `metrics_to_scalars`. But we (I think) want a different error message.

Reply:
That function's message looks generic enough to me so that it works here. Keep in mind the error would be accompanied by a stacktrace. I was proposing `__to_tensor` because it does a few more things

Reply:
I'm ok with reusing/moving metrics_to_scalars. But pretty sure `__to_tensor` is not suitable, because here in fabric, we want to send the result of `.item()` to the logger. Converting first to a tensor and moving to the device is redundant in this case.

Reply:
See 789d413